ddtrace/tracer/rules_sampler.go

```diff
@@ -54,6 +54,55 @@ func (r *rulesSampler) HasSpanRules() bool { return r.spans.enabled() }
 
 func (r *rulesSampler) TraceRateLimit() (float64, bool) { return r.traces.limit() }
 
+type Provenance int32
```

Comment:
```suggestion
type provenance int32
```

I feel we don't need to export this type and related functions.